src/await_async/mini_future.cr

```diff
@@ -5,33 +5,33 @@ class MiniFuture(T)
 
   @status = :running
   @channel = Channel(T).new(1)
-  @error : Exception? = nil
+  @error : Exception?
   @output : T?
 
-  def initialize(prioritary = true, &block : -> T)
+  def initialize(immediate = true, &block : -> T)
     spawn do
       begin
         @channel.send block.call
-      rescue e
-        @error = e
+      rescue ex
+        @error = ex
       ensure
         @status = :terminated
       end
     end
 
-    Fiber.yield if prioritary
+    Fiber.yield if immediate
   end
 
-  def error?
-    @error.any?
+  def running?
+    @status == :running
   end
 
   def finished?
     !running?
   end
 
-  def running?
-    @status == :running
+  def error?
+    @error
```

Comment:
Should be `!!@error` to ensure a boolean output

Reply:
I did it on purpose, since one might want to know the actual error.

Reply:
So it would be great to have 3 methods `error/error!/error?` which return the error or nil, `@error.not_nil` and `!!@error`. 

It will follow the pattern of many shards currently built.

Reply:
I'd suggest just using `getter?` macro (`error?` -> `Exception?`). I don't see any sensible use cases for `error!` and splitting `error` into 2 methods seems pointless since any1 can do `if future.error?`, or even better `if ex = future.error?`

Reply:
All right then. I'm gonna merge as it as of now. 
Indeed it's not a big deal anyway 😉